src/config.rs

```diff
@@ -1590,13 +1590,15 @@ pub struct Font {
     scale_with_dpi: Option<()>,
 }
 
-fn deserialize_scale_with_dpi<'a, D>(_deserializer: D) -> ::std::result::Result<Option<()>, D::Error>
+fn deserialize_scale_with_dpi<'a, D>(deserializer: D) -> ::std::result::Result<Option<()>, D::Error>
     where D: de::Deserializer<'a>
 {
-    panic!(
-        "The `scale_with_dpi` field has been deprecated, \
-         on X11 the WINIT_HIDPI_FACTOR environment variable can be used instead."
-    );
+    // This seems to be necessary in order for the config containing this entry to be parsed properly
+    let _ignored = bool::deserialize(deserializer);
+    eprintln!(
+        "The `scale_with_dpi` setting has been removed, \
+        on X11 the WINIT_HIDPI_FACTOR environment variable can be used instead.");
+    Ok(None)
```

Comment:
To fall in line with the other warnings we have, this warning should be a yellow color.

The comment about the `_ignored` variable also could be improved a bit. If it really is necessary, changing it to `// This is necessary on order for ...` seems to be a better choice.

The warning message itself seems fine to me.

Reply:
Well the other config file warnings just use the `eprintln!` macro (and are not colored). My first pass was to use the `warn!` macro but logging isn't initialized when the config is being parsed. 

As for the comment about _ignored its not super clear why it should be required. All I know is that if that deserialize function doesn't do that then the following error is emitted and the whole configuration file seems to be ignored:

```
problem with config: font: invalid type: boolean `true`, expected field identifier at line 80 column 19; Using default value
Error: problem with config: deserializing from YAML containing more than one document is not supported; Loading default config
```

The serde docs for the attributes don't mention requiring that you actually attempt to deserialize anything: https://serde.rs/field-attrs.html#serdedeserializewith--path

However I wouldn't be surprised if it was expected and just not made explicit in the docs.

Reply:
I updated the comment to say `// This is necessary in order to get serde to complete deserialization of the configuration`

Its probably better than the first pass but still not particularly informative. Without digging into a bunch of serde internals I don't have any other way of knowing why its happening or if its maybe a bug in serde.

Reply:
> Well the other config file warnings just use the eprintln! macro (and are not colored). My first pass was to use the warn! macro but logging isn't initialized when the config is being parsed.

See https://github.com/jwilm/alacritty/blob/master/src/config.rs#L991-L995.

> As for the comment about _ignored its not super clear why it should be required.

Try changing the type to `Option<bool>` instead of `Option<()>`.

Reply:
The warning has been made yellow.

I tried using `Option<bool>` & `bool`. When trying `Option<bool>` I also tried returning `Ok(None)` and `Ok(Some(true))`. Nothing seems to work. If the _ignored line gets commented out the config fails to parse.

Reply:
Would you still prefer an `Option<bool>` over the `Option<()>` even if it makes no functional difference?